packages/gestalt-datepicker/src/DatePicker.js

```diff
@@ -164,14 +164,17 @@ const DatePickerWithForwardRef: React$AbstractComponent<Props, HTMLDivElement> =
   const innerRef = useRef(null);
   useImperativeHandle(ref, () => innerRef.current);
 
+  const [selected, setSelected] = useState<?Date>(dateValue);
   // We keep month in state to trigger a re-render when month changes since height will vary by where days fall
   // in the month and we need to keep the popover pointed at the input correctly
-  const [selected, setSelected] = useState<?Date>(dateValue);
   const [, setMonth] = useState<?number>();
   const [format, setFormat] = useState<?string>();
   const [updatedLocale, setUpdatedLocale] = useState<?string>();
   const [initRangeHighlight, setInitRangeHighlight] = useState<?Date>();
 
+  // TO DO: Ideally this component should be fully controlled using value + onChange, where selected/setSelected are unnecessary.
```

Comment:
Is this essentially a quicker fix until a full re-write/re-factor is possible?

Reply:
@ayeshakmaz Yes, it's a quick fix. Ideally, we would rewrite DatePicker to fully controlled rather than have internal state. This fix works well, but the nicer/elegant fix would require going to all DatePickers and implementing the logic from the onChange to control the Datepicker value. 

This is a fix for the current component state, which we know we have to sit down and reevaluate.